src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Helpers/Emitter/ConfigurationBinder.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    public sealed partial class ConfigurationBindingGenerator
+    {
+        private sealed partial class Emitter
+        {
+            private bool ShouldEmitMethods(MethodsToGen_ConfigurationBinder methods) => (_sourceGenSpec.MethodsToGen_ConfigurationBinder & methods) != 0;
```

Comment:
Moved from main emitter file. New code for the other extension classes have been added to this directory.

Reply:
Since it's a bit hard to distinguish between new code and existing code that was moved, in the future I would recommend splitting the two changes into separate commits or even separate PRs.